docs/proposals/last-operation-description.md

```diff
@@ -0,0 +1,89 @@
+When called this APB module would update a known annotation on the pod ie: ```apb_last_operation```
+with the description provided by the APB developer. This would be part of the 
+[ansible playbook modules](https://github.com/ansibleplaybookbundle/ansible-asb-modules).
+
+In order to collect this information we would use a watch via the Kubernetes client on the pod resource within the temporary namespace [Pod Rest API](https://docs.openshift.com/container-platform/3.5/rest_api/kubernetes_v1.html#list-or-watch-objects-of-kind-pod).
+This would allow us to react to changes made (i.e to the annotations) on Pod Object. Whenever a change occurred, an update to the JobStatus would happen. Once the pod was deleted we would stop the watch on the pod and update the JobStatus with the final ```last_operation``` annotation value.
+
+Sudo code example
+
+```go
+
+wi, err := k8client.CoreV1().Pods(ns).Watch(meta_v1.ListOptions{})
+changes := wi.ResultChan()
+    for ch := range changes{
+		if ch.Type == watch.Modified{
+			...
+		}
+		if ch.Type == watch.Deleted{
+			close(ch)
+			...
+		}
+	}
+
+```
+
+As this would block, it would need to be done in a background go routine. Using a watch in a background routine should allow us to update the JobState independent of the actual execution of the APB.    
+
+
+### Broker changes
+Below are a set of changes that I believe are in line with the current design. The exact implementation would likely differ but the gist would be the same.
+
+- A new field would be added to the JobState type:
+```Description string ``` .
+The string value , added by the APB module and gathered from the pod annotation, would be stored here.
+
+- Modify the existing provision and deprovision subscribers along with the corresponding work messages.
+Only a namespace field would need be added to these messages as the pod name is already present. 
+Inside the subscriber, the go routine to watch the pod and update the JobStatus would start when a new work msg was received.
+This routine would stop once a deleted change was received in the watch.
+
+- Add a new subscriber for bind. Async bind will be part of the service catalog, so having a Status and Description for an async binding will also be needed.
+Adding a new subscriber and work message for binding operations will allow us to update the JobStatus once async binding arrives.
+
+- Modify last operation handler to pull the description out of the Job state and send it back as part of the response.
+
+### APB Module changes
+Add a module that would handle putting the content from a last operation into the correct annotation on the pod.
+
+Something like the following may make sense:
+
+```
+ asb_last_operation:
+   description:"10%: creating deployment and routes"
+```   
+
+***Note not very familiar with how the ansible apb modules wors under the hood so would need some guidance here***
```

Comment:
nit: ~~wors~~works

Reply:
https://github.com/ansibleplaybookbundle/ansible-asb-modules/pull/8 actually has an example for you of using the kubernetes API to create a secret. If you need any more guidance, I think I can help you out.